components/form-elements/button.js

```diff
@@ -29,18 +29,49 @@ import React from "react";
 const commonButtonClasses =
   "flex items-center justify-center border font-semibold leading-none";
 
+/**
+ * Generates a list of tailwind CSS classes for a particular button type when enabled
+ * @param {String} bType Button Type, either primary, secondary, warning, or selected
+ * @returns a list of CSS classes specialized for the button type
+ */
+function enabledButtonTypeClass(bType) {
+  return [
+    `bg-button-${bType}`,
+    `border-button-${bType}`,
+    `text-button-${bType}`,
+    `fill-button-${bType}`,
+  ];
+}
+
+/**
+ * Like `enabledButtonTypeClass` but switching to the `disabled` variant
+ * @param {String} bType Button Type, either primary, secondary, warning, or selected
+ * @param {String} pseudoClass If specified, then this will be prepended to each class
+ * in the list e.g. for "disabled": "bg-button-secondary-disabled" becomes
+ * "disabled:bg-button-secondary-disabled"`
+ * @returns List of disabled variant of the button tailwind CSS classes
+ */
+function disabledButtonTypeClass(bType, pseudoClass = "") {
+  const pc = pseudoClass !== "" ? `${pseudoClass}:` : "";
+  return enabledButtonTypeClass(bType).map((c) => `${pc}${c}-disabled`);
+}
+
```

Comment:
I think “disabled” doesn’t look disabled enough. I feel like I can still click on it, though others might disagree. Might want to get Meenakshi’s opinion too: 
<img width="36" alt="image" src="https://github.com/IGVF-DACC/igvf-ui/assets/1181324/626b64a8-6de6-41cc-8280-57b7c83c54e3">
And in dark mode it looks completely enabled:
<img width="36" alt="image" src="https://github.com/IGVF-DACC/igvf-ui/assets/1181324/fcf5a5be-1113-4495-8cfe-86cc29b74956">

Reply:
<img width="161" alt="Screenshot 2023-08-02 at 10 50 30 PM" src="https://github.com/IGVF-DACC/igvf-ui/assets/1422171/01a506df-0bc0-4186-8572-9de60dcc8105">

oh interesting because this is what it looks like for me

Reply:
The list of [Tailwind CSS core plugins](https://tailwindcss.com/docs/configuration#core-plugins) is really handy because it lets you define custom class names in an easy way. I was hoping it might help for the disabled classes, but it doesn’t look like it does. But that’s what I was talking about last night with the “Tailwind config.”